PropertyExpression.java

```diff
@@ -0,0 +1,99 @@
+package io.github.syst3ms.skriptparser.lang.base;
+
+import io.github.syst3ms.skriptparser.events.TriggerContext;
+import io.github.syst3ms.skriptparser.lang.Expression;
+import io.github.syst3ms.skriptparser.parsing.ParseContext;
+
+import java.util.function.Function;
+
+/**
+ * A base class for expressions that contain general properties.
+ * In English, we can express properties in many different ways:
+ * <ul>
+ *     <li>Mwexim's book</li>
+ *     <li>the book of Mwexim</li>
+ * </ul>
+ * This utility class will make sure you won't need to write multiple patterns each time
+ * and ensures you can easily handle all these properties, using the {@link #setOwner(Expression)}
+ * and {@link #getOwner()} methods to handle the owner.
+ *
+ * The class also has a built-in {@link #init(Expression[], int, ParseContext)} and {@link #getValues(TriggerContext)},
+ * ensuring you only have to do the bare minimum. These methods are very basic though, so most of the time,
+ * you'll want to override them anyway.
+ *
+ * @param <T> The returned type of this expression.
+ * @param <O> The type of the owner of this expression.
+ */
+public abstract class PropertyExpression<T, O> implements Expression<T> {
+    private Expression<O> owner;
+
+    public Expression<O> getOwner() {
+        return owner;
+    }
+
+    public void setOwner(Expression<O> owner) {
+        this.owner = owner;
+    }
+    
+    /**
+     * If you're property only relies on one simple method, you can define that method here
+     * using a {@link Function}. This function will be applied at the {@link #getValues(TriggerContext)} method
+     * of this class.
+     * @return the function that needs to be applied in order to get the correct values.
+     */
+    public Function<O[], T[]> getPropertyFunction() {
+        return null;
+    }
+
+    /**
+     * There are 2 kinds of possession:
+     * <ul>
+     *     <li><b>Genitive:</b>Mwexim's book</li>
+     *     <li><b>Regular:</b>book of Mwexim</li>
+     * </ul>
+     * Because both have different patterns and the order of the expressions can be important,
+     * you can use this method to check if the pattern is the Genitive form. If it's not, it's the regular form.
+     * @param matchedPattern the matched pattern of a property
+     * @return whether this pattern was the genitive form (true) or the regular form (false)
+     */
+    public static boolean isGenitive(int matchedPattern) {
+        return matchedPattern == 0;
+    }
+
+    /**
+     * This {@code init()} method overrides the default method and does the initialization by itself.

```

Comment:
```suggestion
     * This default {@code init()} implementation automatically properly sets the owner of this property,
     * which can be accessed using {@link #getOwner()}. If this implementation is overridden for one reason
     * or another, it must call {@link #setOwner(Expression<O>)} properly.
```
I think this specification would be more appropriate.